macOS/DuckDuckGo/Permissions/Model/PermissionModel.swift

```diff
@@ -130,7 +142,22 @@ final class PermissionModel {
                     permissions.geolocation
                         .systemAuthorizationDenied(systemWide: !geolocationService.locationServicesEnabled())
                 } else {
-                    permissions.geolocation.update(with: webView.geolocationState)
+                    let currentState = webView.geolocationState
+
+                    // Track if geolocation was ever activated
+                    if currentState == .active {
+                        geolocationWasActivated = true
+                    }
+
+                    // With new permission view, keep geolocation as active once it's been used
+                    if featureFlagger.isFeatureOn(.newPermissionView),
+                       geolocationWasActivated,
+                       currentState == .none,
+                       permissions.geolocation == .active || permissions.geolocation == .inactive {
+                        permissions.geolocation = .active
```

Comment:
`geolocationWasActivated` seems redundant here, since there‘s the `inactive` state - when geolocation was queried once, or activated, granted and then deactivated, it falls to the `inactive` state for the icon to stay in the permissions, so  `geolocationWasActivated` is defined by `permissions.geolocation == .active || permissions.geolocation == .inactive`